src/Symfony/Bridge/Twig/Extension/IntlExtension.php

```diff
@@ -0,0 +1,317 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Twig\Extension;
+
+use Symfony\Component\Intl\DateFormatter\IntlDateFormatter;
+use Symfony\Component\Intl\NumberFormatter\NumberFormatter;
+use Symfony\Component\Intl\Intl;
+
+/**
+ * Provides integration of the Intl component with Twig.
+ *
+ * @author Jules Pietri <jules@heahprod.com>
+ */
+class IntlExtension extends \Twig_Extension
+{
+    private $defaultLocale;
+    private $dateFormatterClass;
+    private $numberFormatterClass;
+    private $dateFormatters = array();
+    private $dateFormatValues = array();
+    private $numberFormatters = array();
+    private $numberTypeValues = array();
+    private $numberStyleValues = array();
+
+    public function __construct($defaultLocale = null)
+    {
+        $this->defaultLocale = $defaultLocale ?: \Locale::getDefault();
```

Comment:
you should not read ``Locale::getDefault()`` at instantiation time, but when using it, as the Intl default locale can change after instantiating Twig (Symfony will change it at runtime when determining the locale based on the routing)